src/Components/RegistrationForm.jsx

```diff
@@ -0,0 +1,307 @@
+import {
+  faCheck,
+  faTimes,
+  faEye,
+  faEyeSlash,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import "../Registration.css";
+import { useState, useEffect } from "react";
+import HorizontalLine from "./HorizontalLine";
+import PrimaryButton from "./PrimaryButton";
+import CloseIcon from "./CloseIcon";
+import { Link } from "react-router-dom";
+import { useNavigate } from "react-router-dom";
+
+const RegistrationForm = () => {
+  const navigate = useNavigate();
+  const EMAIL_REGEX = /^[^\s@]+@[^\s@]+\.[^\s@]+$/; // A simple check, e-mail might still not be valid.
+  const PASSWORD_REGEX =
+    /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%?&_.,:;"'~=+-/|\\{}()^\[\]]).{8,24}$/; // Password requirements:
+  // between 8-24 characters and includes:
+  // a lowercase letter
+  // an uppercase letter
+  // a number
+  // a special character.
+
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
+
+  const [email, setEmail] = useState("");
+  const [validEmail, setValidEmail] = useState(false);
+
+  const [password, setPassword] = useState("");
+  const [validPassword, setValidPassword] = useState(false);
+
+  const [confirmPassword, setConfirmPassword] = useState("");
+  const [passwordMatch, setPasswordMatch] = useState(false);
+
+  const [termsAccepted, setTermsAccepted] = useState(false);
+  const [errorMsg, setErrorMsg] = useState("");
+
+  const [hidePassword, setHidePassword] = useState(true);
+  const [hidePassword2, setHidePassword2] = useState(true);
+
+  const [users, setUsers] = useState([]);
+
+  const handleRegister = (e) => {
+    e.preventDefault();
+    if (!validEmail || !validPassword || !passwordMatch || !termsAccepted) {
+      setErrorMsg("Please fill out all fields correctly.");
+      return;
+    }
+    setErrorMsg("");
+
+    // Handle form submission - needs to be implemented with backend.
+    {
+      console.log(firstName);
+    }
+    {
+      console.log(lastName);
+    }
+    const newUser = {
+      firstName: firstName,
+      lastName: lastName,
+      email: email,
+      password: password,
+    };
+
+    setUsers((prevUsers) => [...prevUsers, newUser]);
```

Comment:
Ah? Why would a registration form have previous users. I don't understand the logic